openquake/commonlib/readinput.py

```diff
@@ -728,9 +748,11 @@ def read(cls, fname, calculation_mode='', insured_losses=False,
         param['ignore_missing_costs'] = set(ignore_missing_costs)
         exposure, assets = _get_exposure(
             param['fname'], param['all_cost_types'])
-        nodes = assets if assets else exposure._read_csv(
-            ~assets, os.path.dirname(param['fname']))
-        exposure._populate_from(nodes, param)
+        _nodes = assets if assets else exposure._read_csv(
+            assets.text, os.path.dirname(param['fname']))
+        if nodes:  # this is useful for Paul Henshaw
+            return nodes
```

Comment:
I think you mean return _nodes here.
Also I suggest changing the comment to "This is useful for the GED4ALL import script"

Reply:
Done. BTW I have changed the signature of the call, you should use `Exposure.read(fname, asset_nodes=True)` which I believe is clearer.